### ISSUE TITLE
"Plain" train mode still build the oblivious tree

### ISSUE THREAD

**Issue by ccylance** [catboost/catboost#2628]
catboost version: catboost 1.2
Operating System: macos 14.4.1
CPU: M1
GPU: no

According to the paper, it said the plain mode will build the tree by following the GBDT strategy. However, when I plot the tree trained with Plain mode, the structure of tree is oblivious tree not the decision tree. 
Another confusing point is why the priornum used in each layer is different？
One more thing is how the leaf value calculated? 
![image](https://github.com/catboost/catboost/assets/54780246/78ae0973-ae50-4319-a275-d3da60698b14)
![image](https://github.com/catboost/catboost/assets/54780246/e89f26f7-5d34-4a21-b3c6-09451b4ad958)

**Comment by ccylance** [comment]
> > According to the paper, it said the plain mode will build the tree by following the GBDT strategy. However, when I plot the tree trained with Plain mode, the structure of tree is oblivious tree not the decision tree.
> 
> Oblivious trees are a subtype of decision trees so no contradiction here. If you want CatBoost to build non-oblivious trees you can specify [`grow_policy` training parameter](https://catboost.ai/en/docs/references/training-parameters/common#grow_policy) as `Depthwise` or `Lossguide`.
> 
> > Another confusing point is why the priornum used in each layer is different？
> 
> Because there can be [several numerical features derived from categorical ones](https://catboost.ai/en/docs/concepts/algorithm-main-stages_cat-to-numberic) with different priors.
> 
> You can see the default sets of priors for each Ctr type [here](https://github.com/catboost/catboost/blob/1b47df44b3653c6535ddf1629ba5daa58ba9dd94/catboost/private/libs/options/cat_feature_options.cpp#L118-L138).
> 
> It is also possible [to specify your own priors using training parameters](https://catboost.ai/en/docs/references/training-parameters/ctr).

Appreciate for your help. One more thing is how the leaf value calculated? Is it the mean of the residuals of all items allocated to the leaves?

**Comment by andrey-khropov** [comment]
> how the leaf value calculated? In the tree_plot it shows as 0.071 but the prediction return the value 0.5166, and I am confused about how the leaf node value is converted into the predicted value？

1) You display only the single tree (specified by `tree_idx`), but the model typically contains many trees (up to the number of [`iterations`](https://catboost.ai/en/docs/references/training-parameters/common#iterations) but may have less than the specified maximum value of iterations if [the overfitting detector](https://catboost.ai/en/docs/features/overfitting-detector-desc) and/or [`use_best_model`](https://catboost.ai/en/docs/references/training-parameters/common#use_best_model) are used). The number of trees in the model can be obtained from [the `tree_count_` attribute](https://catboost.ai/en/docs/concepts/python-reference_catboostclassifier_attributes#tree_count_) of the trained model.

2) [The return value](https://catboost.ai/en/docs/concepts/python-reference_catboostclassifier_predict#output-format) of [`predict`](https://catboost.ai/en/docs/concepts/python-reference_catboostclassifier_predict) will be the function of the sum of values of the selected leaves in all trees. Post-transformation depends on the specified [`prediction_type` parameter](https://catboost.ai/en/docs/concepts/python-reference_catboostclassifier_predict#prediction_type).

**Comment by ccylance** [comment]
> 2\. `prediction_type` parameter

In order to find the relationship between the leaf value and prediction result, I only trained 1 tree but they are still different
